testConnectivity.py

```diff
@@ -418,66 +423,110 @@ def testConnectivity(args):
+                    bsum = sum(phaseCounts.take(brange, mode='wrap')) # backward sum
+                    if fsum > bsum:
+                        phase2Write = int((badPhaseCounts[0]+WINDOW)%GBT_PHASE_RANGE)
+                    elif bsum > fsum:
+                        phase2Write = int((badPhaseCounts[0]-WINDOW)%GBT_PHASE_RANGE)
+                    else:
+                        ## choose the phase that doesn't require passing 15?
+                        phase2Write = int((badPhaseCounts[0]-WINDOW)%GBT_PHASE_RANGE)
+                    pass
+                elif len(badPhaseCounts) == 2:
+                    # exactly two bad phases, pick the midpoint, don't worry about wraparound
+                    # shoulw we check the distance?
+                    phase2Write = int((badPhaseCounts[1] - badPhaseCounts[0])/2+badPhaseCounts[0])
+                    pass
+                elif len(badPhaseCounts) == 3:
+                    # check if bad phases are sequential, if so use pick the midpoint, ignore wraparound
+                    # if bad phases are not sequential just look for the longest good window
+                    badPhasesAreSequential=False
+                    minSeqPhase = -1
+                    maxSeqPhase = -1
+                    idx2Use = [ 0, 1, 2]
+                    for idx1 in range(0,3):
+                        for idx2 in range(0,3):
+                            if not (idx1 > idx2):
+                                continue
+                            if( int(abs(badPhaseCounts[idx1]-badPhaseCounts[idx2])) == 1): #bad phases are sequential
+                                minSeqPhase = int(min(badPhaseCounts[idx1],badPhaseCounts[idx2]))
+                                maxSeqPhase = int(max(badPhaseCounts[idx1],badPhaseCounts[idx2]))
+                                idx2Use.remove(idx1)
+                                idx2Use.remove(idx2)
+                                badPhasesAreSequential=True
+                                break # exit inner loop
+                            pass
+                        if (badPhasesAreSequential):
+                            break # exit outer loop
+                        pass
+                    if (badPhasesAreSequential): # Look for midpoint
+                        if badPhaseCounts[idx2Use[0]] > maxSeqPhase:
+                            phase2Write = int((badPhaseCounts[idx2Use[0]] - maxSeqPhase)/2+maxSeqPhase)
+                        else:
+                            phase2Write = int((minSeqPhase - badPhaseCounts[idx2Use[0]])/2+badPhaseCounts[idx2Use[0]])
+                            pass
                         pass
-                    pass # End loop over phases
+                    else:                       # Look for longest good window
+                        badPhaseCounts = np.sort(badPhaseCounts)
+                        ranges = []
+                        ranges.append(crange(0,
```

Comment:
don't need `crange` here, as you're explicitly constructing ranges that will never wrap

Reply:
Why invent another function when an existing one will do?

Reply:
using `range` is not inventing something ;-)

Reply:
Hah, okay fair point.  I'll implement that.

Reply:
When reviewing the 3 bad phases steps, I was thinking that simply finding the distance between each pair would be simpler.

Say you have the following (taken from your real world example)
```
badPhases1 = [2,9,10]
badPhases2 = [2,8,10]
```
What phase would you like to choose?
In the current algorithm, you'd take `goodPhase1 = (9-2)/2+2`, and in the second you'd take `goodPhase2 = (8-4)` or `goodPhase2 = (10+4)`, right?
If you take the difference between successive elements, you'd end up with
```
deltaPhases1 = [7, 1, 8]
deltaPhases1 = [6, 2, 8]
```
If you then take the points that have the largest delta, and apply the case 1 algorithm (count both ways for max good phases), you'd wind up with `goodPhase1 = 10+4` and `goodPhase2 = 10+4` (though maybe one wouldn't want to take this phase, as it sits right next to 15 (are we choosing 14 as the phase for any VFATs that fall into the other cases currently?)

My motivation for this is that it's much simpler than the various cases

Reply:
Here is what I used to test this:
```python
ar1 = np.array([2,9,10])
ar2 = np.array([2,8,10])
ar1p = np.array([ar1[(x+1)%len(ar1)] for x in range(len(ar1)) ])
ar2p = np.array([ar2[(x+1)%len(ar2)] for x in range(len(ar2)) ])
ar1res = (ar1p-ar1)%16
ar2res = (ar2p-ar2)%16
ar1res
ar2res
```